src/QsCompiler/LanguageServer/EditorState.cs

```diff
@@ -153,6 +153,33 @@ internal bool QsProjectLoader(Uri projectFile, out ProjectInformation info)
             return true;
         }
 
+        internal bool QsTemporaryProjectLoader(Uri sourceFileUri, string sdkVersion, out Uri projectUri, out ProjectInformation info)
+        {
+            projectUri = null;
+            info = null;
+
+            var localFolderPath = Path.GetDirectoryName(sourceFileUri.LocalPath);
+            var projectFolderPath = Directory.CreateDirectory(Path.Combine(Path.GetTempPath(), "qsharp", $"{localFolderPath.GetHashCode():X8}")).FullName;
```

Comment:
NIT: Any reason to use upper case instead of lower case here?

Reply:
No reason.. 😊 do you have a preference?

Reply:
As a preference, I think that filenames should be in lower case, but this is unimportant.

Reply:
Sounds reasonable, I'll flip it to lower case.